include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -505,10 +484,25 @@ class fm_index
         text_begin_ss.set_vector(&text_begin);
         archive(text_begin_rs);
         text_begin_rs.set_vector(&text_begin);
+
+        auto sigma = alphabet_size<alphabet_t>;
         archive(sigma);
-        bool tmp = is_collection_;
+        if (sigma != alphabet_size<alphabet_t>)
+        {
+            throw std::logic_error{"The fm_index was built over an alphabet of size " + std::to_string(sigma) +
+                                   " but it is being read into a fm_index with an alphabet of size " +
+                                   std::to_string(alphabet_size<alphabet_t>) + "."};
```

Comment:
Just some random thoughts:

We don't have any real check wether the target and source alphabet are the same.

So the user must ensure that the alphabet type stays correct. And we can't infer it from the index itself.

So `rna4` and `dna4` would be interchangeable, because they have the same alphabet_size. The same with  `gapped<dna4>` and `dna5`. I think this is completely fine.

Reply:
Yup, we can't make sure it's the same alphabet, but at least we can do some checks.